cpu/kinetis/periph/gpio.c

```diff
@@ -86,15 +86,14 @@
 
 #ifdef MODULE_PERIPH_GPIO_IRQ
 /**
- * @brief   Calculate the needed memory (in byte) needed to save 4 bits per MCU
- *          pin
+ * @brief   Calculate the needed memory (in bytes) to store 4 bits per MCU pin
  */
-#define ISR_MAP_SIZE        (GPIO_PORTS_NUMOF * PINS_PER_PORT * 4 / 8)
+#define ISR_MAP_SIZE (GPIO_PORTS_NUMOF * PINS_PER_PORT * 4 / 8 / sizeof(uint32_t))
```

Comment:
So just for my understanding: you can have 16 interrupts in total, mapped to any pin?

Reply:
Yes, more or less. We can have up to all pins mapped, with up to `CTX_NUMOF` unique callbacks to map them to.

`isr_map` is what maps a pin to (if applicable) a callback context in `isr_ctx`

In `isr_map` each pin gets 4 bits to specify an `isr_ctx` which is 16 minus 1 possible callbacks that a pin can point to. It's minus 1 because we need a value to indicate that there is no `isr_ctx` for that pin in `isr_map` and that value is 0. There used to be no concept of a no-context value in `isr_map` and that's what the `fix irq contexts not being freed` commit is for.

An additional limitation is `CTX_NUMOF` which currently limits us to only 8 callback contexts in `isr_ctx`. We could increase this to 15 before having to change the workings of `isr_map`.